libraries/terminal/src/ITerminal.ts

```diff
@@ -2,7 +2,6 @@
 // See LICENSE in the project root for license information.
 
 import type { ITerminalProvider } from './ITerminalProvider';
-import type { IColorableSequence } from './Colors';
 
 /**
```

Comment:
@theJiawen encountered an interesting problem in PR 4389:

https://github.com/microsoft/rushstack/pull/4389/files#diff-cea60748a6a3f19d7221df346cedad9f5515ab5b1595354cffae5f3fb5a2b8d8

In his PR, Rush is intercepting the output of `pnpm install` to scan for custom tips, which means the STDIO is `pipe` instead of `inherit`. This removes the isTTY attribute from PNPM's `stdout` which causes it to wrongly infer that colors are not supported.  The PR solves this problem by assigning `FORCE_COLOR` based on Rush's own computation of whether colors are supported.

In the current design, `ITerminal` is the thing that we pass around everywhere, but it does not know whether colors are supported.  (In fact, Rush's `ITerminal` prints to both non-colorable log file and a possibly colorable TTY.)  How should we solve this problem with the new API?

Reply:
The `Terminal` object strips out colors when writing to terminal providers that don't support them, regardless of where the colors come from.